modules/auth/proc-adding-oauth2-proxy-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,83 @@
+= Adding OAuth2 proxy as an authentication provider in {product}
+
+The {product} `core-plugin-api` package includes an OAuth2-proxy authentication provider
+that can authenticate users
+by using an link:https://github.com/oauth2-proxy/oauth2-proxy[OAuth2 proxy] in front of an {product} instance.
+This enables to reuse existing authentications within a cluster.
+In general, the OAuth2 proxy supports all OpenID Connect providers.
+For more details, see link:https://oauth2-proxy.github.io/oauth2-proxy/configuration/providers/[the supported provider list].
+
+.Procedure
+
+. Set up the OAuth2-proxy authentication provider in the {product} backend (`app-config.yaml`).
++
+Add the `oauth2Proxy` provider configuration to your `app-config.yaml` under the root `auth` configuration:
++
+[source,yaml]
+----
+auth:
+  environment: development
+  providers:
+    oauth2Proxy: {}
+----
++
+The OAuth2-proxy provider is a structure with no configuration keys.
+However, it requires the empty object to enable the provider in the auth backend.
+
+. To use the oauth2Proxy provider you must also configure it with a sign-in resolver.
++
+For the oauth2Proxy provider, the sign-in result is quite different than other providers.
+Because it's a proxy provider that can be configured to forward information through arbitrary headers, the auth result simply just gives you access to the HTTP headers of the incoming request.
+Using these you can either extract the information directly, or grab ID or access tokens to look up additional information and/or validate the request.
++
+A simple sign-in resolver might for example look like this:
++
+.`packages/backend/src/plugins/auth.ts`
+[source,javascript]
+----
+providerFactories: {
+  ...defaultAuthProviderFactories,
+  oauth2Proxy: providers.oauth2Proxy.create({
+    signIn: {
+      async resolver({ result }, ctx) {
+        const name = result.getHeader('x-forwarded-user');
+        if (!name) {
+          throw new Error('Request did not contain a user')
+        }
+        return ctx.signInWithCatalogUser({
+          entityRef: { name },
+        });
+      },
+    },
+  }),
+},
+----
+
+. (Optionally) To enable the OAuth2-proxy provider for sign-in, add the provider to the {product} front-end (`packages/app/src/App.tsx`).
++
+To add the provider to the front-end, add the
+`ProxiedSignInPage` component to the sign-in page.
++
+To configure sign-in, provide a custom `SignInPage` app component.
+The app renders this component before any other routes.
+The component provides the identity of the current user.
+The `SignInPage` can render any number of pages and components, or just blank space with logic running in the background.
+In the end however it must provide a valid {product} user identity through the `onSignInSuccess` callback prop, at which point the rest of the app is rendered.
++
+The following example for OAuth2-proxy shows the additions needed to `packages/app/src/App.tsx`:
```

Comment:
In RHDH container it is not possible to edit files like this, there needs to be another way to configure this